### PR TITLE
Add commit message rules in the CONTRIBUTING file

### PR DESCRIPTION
**Pull request by trufae**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

...

**Closing issues**

...

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -71,6 +71,20 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+
+* Start the message capitalized (first letter in uppercase)
+* Be short and concise, the whole concept must fit one line
+* If a command is inlined, use backticks
+* Add a double-hashtag if the change matters for the changelog
+* Search the changelog to known hashtags (see sys/release-notes.sh)
+* If the commit fixes a bug start with 'Fix #number - '
+* For extra details, add an empty line and use asterisk item list below
+* Use present (Add vs Added, Fix vs Fixed/Fixes)
+* If the PR have multiple isolable changes squash/rebase them and inform it in the PR
```
Comment (ret2libc) [inline CONTRIBUTING.md:87]: what does this mean? It's not clear to me.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -71,6 +71,20 @@ When afraid to touch force and risk losing your work (do backups!..),
...
 
+## Commit message rules
+
+When commiting your changes into the repository you may want to follow some
+rules to make the git history more readable and consistent:
+
+* Start the message capitalized (first letter in uppercase)
+* Be short and concise, the whole concept must fit one line
+* If a command is inlined, use backticks
+* Add a double-hashtag if the change matters for the changelog
```
[the remark this answers is not in the document]
Comment (XVilka) [reply CONTRIBUTING.md:82]: Not "matters", maybe something like "to assign the change to the corresponding section in the changelog"?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -71,6 +71,20 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+* Search the changelog to known hashtags (see sys/release-notes.sh)
+* If the commit fixes a bug start with 'Fix #number - '
+* For extra details, add an empty line and use asterisk item list below
+* Use present (Add vs Added, Fix vs Fixed/Fixes)
```
Comment (XVilka) [inline CONTRIBUTING.md:86]: "Present Simple grammar tense"

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -71,6 +71,20 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+## Commit message rules
+
+When commiting your changes into the repository you may want to follow some
+rules to make the git history more readable and consistent:
+
+* Start the message capitalized (first letter in uppercase)
+* Be short and concise, the whole concept must fit one line
+* If a command is inlined, use backticks
+* Add a double-hashtag if the change matters for the changelog
+* Search the changelog to known hashtags (see sys/release-notes.sh)
```
Comment (XVilka) [inline CONTRIBUTING.md:83]: I don't think this entry is necessary.

File: CONTRIBUTING.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kazarmy) [reply CONTRIBUTING.md:82]: There should be a complete list of hashtags.

File: CONTRIBUTING.md (hunk 1 above)
Comment (trufae) [reply CONTRIBUTING.md:87]: If the pr contains multiple commits that are not suposed to be squashed it may state that properly and squash the commits sccordingly

File: CONTRIBUTING.md (hunk 3 above)
Comment (trufae) [reply CONTRIBUTING.md:86]: Please submit the suggestion so i can merge the changr

File: CONTRIBUTING.md (hunk 4 above)
Comment (trufae) [reply CONTRIBUTING.md:83]: Maybe reword to: hashtags are used by the release notes script

File: CONTRIBUTING.md (hunk 1 above)
Comment (XVilka) [reply CONTRIBUTING.md:87]: I think this should be reworded then, indeed it's unclear what does it mean.

File: CONTRIBUTING.md (hunk 4 above)
Comment (ret2libc) [reply CONTRIBUTING.md:83]: I'm not sure new contributors would care to know.

Comment (XVilka) [reply CONTRIBUTING.md:83]: It's wrong to ask new developers to search something. The all available hashtags should be listed here, in this document, otherwise they can't be made mandatory.

Comment (trufae) [reply CONTRIBUTING.md:83]: Feel free to commit a list of tags or do you prefer a shell oneliner that prints them all? This pr is suposed to be collaborative. So feel free to submit suggestion changes

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -71,6 +71,40 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+* ##egg      - the `r_lang` compiler
+* ##io       - related to the `r_io` library
+* ##lang     - bindings
+* ##port     - portability (new OS/archs)
+* ##projects - saving/loading state
+* ##refactor - improve code quality
+* ##search   - rafind2, / command, ..
+* ##shell    - commandline, newshell, ..
+* ##signatures-searching/generating them
+* ##util     - core apis
```
Comment (kazarmy) [inline CONTRIBUTING.md:107]:
```suggestion
* ##anal     - analysis related
* ##asm      - assembler
* ##bin      - binary parsing
* ##build    - build fixes/changes
* ##config   - config var changes/additions/renamings
* ##cons     - console/terminal-related
* ##crypto   - cryptography
* ##debug    - debugger stuff
* ##diff     - diffing code, strings, basic blocks, ...
* ##disasm   - disassembler
* ##doc      - documentation
* ##egg      - the `r_lang` compiler
* ##emu      - emulation, including esil
* ##graph    - basic block graph, callgraph, ...
* ##io       - related to the `r_io` library
* ##json     - json changes/fixes
* ##lang     - bindings
* ##meta     - metadata handling other than printing
* ##optimization-space/time optimizations
* ##port     - portability (new OS/archs)
* ##print    - printing data, structures, strings, tables, types ..
* ##projects - saving/loading state
* ##refactor - improve code quality
* ##remote   - r2 over tcp, http, rap, serial .. including collaboration
* ##search   - rafind2, / command, ..
* ##shell    - commandline, newshell, ..
* ##signatures-searching/generating them
* ##test     - testing infrastructure changes/fixes
* ##util     - core apis
* ##visual   - visual ui, including panels
```

Comment (trufae) [reply CONTRIBUTING.md:107]: Use the pencil

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -71,6 +71,53 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+rules to make the git history more readable and consistent:
+
+* Start the message capitalized (only the first character must be in uppercase)
+* Be short and concise, the whole concept must fit one line
+* If a command is inlined, use backticks
+* Add a double-hashtag if the change matters for the changelog (See below)
+* If the commit fixes a bug start with 'Fix #number - '
+* For extra details, add an empty line and use asterisk item list below
+* Use present simple grammar tense (Add vs Added, Fix vs Fixed/Fixes)
+* Prefix with `WIP:` if it's not yet ready to be merged
```
Comment (ret2libc) [inline CONTRIBUTING.md:86]: Isn't this just something about PRs and not about `Commit message rules`?

Comment (kazarmy) [reply CONTRIBUTING.md]: Yes 13b8cc4a0067cc17b08d11ae403bcc44d0a8ee50

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -71,6 +72,52 @@ When afraid to touch force and risk losing your work (do backups!..),
...
+## Commit message rules
+
+When commiting your changes into the repository you may want to follow some
+rules to make the git history more readable and consistent:
+
+* Start the message capitalized (only the first character must be in uppercase)
+* Be short and concise, the whole concept must fit one line
+* If a command is inlined, use backticks
+* Add a double-hashtag if the change matters for the changelog (See below)
+* If the commit fixes a bug start with 'Fix #number - '
```
Comment (ret2libc) [inline CONTRIBUTING.md:84]: Actually... What about putting the `Fix #number` in the third line? I think it becomes hard to put a concept in one single line if you have `Fix #XXXXX - something (#123131312) ##sometag`. There are already way too many things IMHO. WDYT?

Comment (XVilka) [reply CONTRIBUTING.md:84]: Agree, it makes a better sense. We could even use the same approach as with "Signed-off: bla-bla" - just put it on the last line.

Comment (trufae) [reply CONTRIBUTING.md:84]: Multiline commits are a pain to read and parse. If the info is not relevant or its just detailed info about the first line lgtm. But the first line should describe everything contained in the commit without going too deep into details.

Comment (kazarmy) [reply CONTRIBUTING.md:84]: I think the `Fix #number` should be in the first line. We want to encourage people to fix issues (this is monetarily-unpaid technical support after all) and having the fixes prominently displayed in the commit list is, imho, one small way of saying thank you.

Besides, it's not really a problem to have an overly long first line that just spills over into a second line.

Comment (ret2libc) [reply CONTRIBUTING.md:84]: Actually it is a problem, because the first line can only be 72 (or  80, I don't remember) chars long and it is the "title" of the commit/PR (what you see when you use `git log --oneline`, for example). I'm ok with keeping it in first line as well though.

Comment (trufae) [reply CONTRIBUTING.md:84]: i dnt think gh obeys the Fix # in any other line, but never tried
